test/hotspot/jtreg/compiler/vectorapi/reshape/utils/TestCastMethods.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorapi.reshape.utils;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static compiler.vectorapi.reshape.utils.VectorReshapeHelper.*;
+import static compiler.vectorapi.reshape.utils.VectorSpeciesPair.makePair;
+
+/**
+ * The cast intrinsics implemented on each platform, commented out tests are the ones that are
+ * supposed to work but currently don't.
+ */
+public class TestCastMethods {
+    public static final List<VectorSpeciesPair> AVX1_CAST_TESTS = List.of(
+            makePair(BSPEC64, SSPEC64),
+            makePair(BSPEC64, SSPEC128),
+            makePair(BSPEC64, ISPEC128),
+            makePair(BSPEC64, FSPEC128),
+            // makePair(BSPEC64, DSPEC256),
```

Comment:
May later we consider a negative test that we don't expect the required IR node(s). I am uncertain in these cases whether it's an implementation restriction or a hardware restriction, if the former when the restriction goes away the test will fail and we update it.